glean-core/src/database/mod.rs

```diff
@@ -16,6 +19,21 @@ use crate::Glean;
 use crate::Lifetime;
 use crate::Result;
 
+lazy_static!{
+    // A map of all the known 'user' lifetime metrics along with their expiration
+    // dates. See bug 1604854 for more context on why this is needed.
+    static ref USER_LIFETIME_EXPIRATION_MAP: HashMap<&'static str, NaiveDate> = vec![
+        ("glean_internal_test.user_metric_expired", NaiveDate::from_ymd(2015, 3, 14)),
```

Comment:
After the design questions are settled, I'll add the metrics as a separate PR.

Reply:
@badboy is there a bug for this, already?

Reply:
No, but I can file one.

Reply:
filed: https://bugzilla.mozilla.org/show_bug.cgi?id=1626558